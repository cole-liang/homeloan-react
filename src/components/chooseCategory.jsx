import React, { Component } from "react";

import { connect } from "react-redux";
import { Row, Col } from "react-bootstrap";
import { ReactComponent as CoupleSVG } from "../assets/couple.svg";
import { ReactComponent as SingleSVG } from "../assets/single.svg";

import styled from "styled-components";

const ChooseCategoryDiv = styled.div`
  & .coupleSvg {
    position: relative;
    top: -6px;
  }

  & [id$="Icon"] {
    cursor: pointer;
    transition: all 0.6s;
  }

  & [id$="Icon"]:hover {
    transform: scale(1.1);
  }
`;

export class ChooseCategory extends Component {
  handleClick = category => {
    this.props.onClick(category);
  };

  render() {
    const { user } = this.props;
    const welcomeMsg = !!user
      ? `Welcome ${user.firstName}! Choose one to get you on board:`
      : "Welcome! Choose one to get you on board:";

    return (
      <ChooseCategoryDiv className="d-flex flex-column justify-content-center flex-grow-1">
        <Row noGutters className="noPadding">
          <div className="welcomeText my-3">{welcomeMsg}</div>
        </Row>
        <Row noGutters className="noPadding">
          <Col xs={12} md={6}>
            <Row noGutters className="justify-content-center my-4">
              <div id="singleIcon" onClick={() => this.handleClick("single")}>
                <SingleSVG className="singleSvg" />
                <div className="iconDesc text-center">Single</div>
              </div>
            </Row>
          </Col>
          <Col xs={12} md={6}>
            <Row noGutters className="justify-content-center my-4">
              <div id="coupleIcon" onClick={() => this.handleClick("couple")}>
                <CoupleSVG className="coupleSvg" />
                <div className="iconDesc text-center">Couple</div>
              </div>
            </Row>
          </Col>
        </Row>
      </ChooseCategoryDiv>
    );
  }
}

const mapStateToProps = state => {
  return {
    user: state.user
  };
};

export default connect(
  mapStateToProps,
  null
)(ChooseCategory);
