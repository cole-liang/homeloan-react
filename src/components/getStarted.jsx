import React, { Component } from "react";
import ChooseCategory from "./chooseCategory";
import LoanInfoForm from "./loanInfoForm";
import Result from "./result";

import { Steps } from "antd";
import { Row, Col } from "react-bootstrap";

import styled from "styled-components";

const GetStartedDiv = styled.div`
  background: #fff;
  height: 100%;

  & .welcomeText {
    font-size: 25px;
  }

  /* Resolve conflict between Ant Design and Bootstrap*/
  & .steps svg {
    vertical-align: inherit;
  }
`;

const Step = Steps.Step;

class GetStarted extends Component {
  state = { currentStep: 2, category: "", canHelp: false };

  previousStep = currentStep => {
    this.setState({ currentStep: currentStep - 1 });
  };

  nextStep = currentStep => {
    this.setState({ currentStep: currentStep + 1 });
  };

  handleCategory = category => {
    const { currentStep } = this.state;
    this.setState({ category });
    this.nextStep(currentStep);
  };

  handleSubmit = data => {
    const { currentStep } = this.state;
    const canHelp = this.calculateProb(data);
    console.log(data, canHelp);
    this.setState({ canHelp }, () => {
      this.nextStep(currentStep);
    });
  };

  calculateProb = ({
    yourIncome,
    yourExpense,
    partnerIncome,
    partnerExpense,
    propertyValue,
    deposit
  }) => {
    const x1 =
      Number.parseFloat(yourExpense) +
      (partnerExpense ? Number.parseFloat(partnerExpense) : 0);
    const x2 =
      Number.parseFloat(yourIncome) +
      (partnerIncome ? Number.parseFloat(partnerIncome) : 0);
    const x = x1 / x2;

    const y = Number.parseFloat(deposit) / Number.parseFloat(propertyValue);

    return x <= 0.5 && y >= 0.2;
  };

  render() {
    const { currentStep, category, canHelp, data } = this.state;
    return (
      <GetStartedDiv>
        <Row noGutters className="noPadding justify-content-center py-5 h-100">
          <Col xs={11} md={9} className="d-flex flex-column">
            <Steps className="steps mb-3" current={currentStep}>
              <Step key="choose" title="Choose" />
              <Step key="basicInfo" title="Basic Info" />
              <Step key="result" title="Result" />
            </Steps>
            <div className="d-flex flex-column justify-content-center flex-grow-1">
              {currentStep === 0 && (
                <ChooseCategory onClick={this.handleCategory} />
              )}
              {currentStep === 1 && (
                <LoanInfoForm
                  isCouple={category === "couple"}
                  previousStep={() => this.previousStep(currentStep)}
                  onSubmit={this.handleSubmit}
                />
              )}
              {currentStep === 2 && <Result canHelp={canHelp} />}
            </div>
          </Col>
        </Row>
      </GetStartedDiv>
    );
  }
}

export default GetStarted;
