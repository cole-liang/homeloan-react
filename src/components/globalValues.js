export const BACKGROUND_URL =
  "http://www.zinqsolutions.com.au/wp-content/uploads/2018/12/zinq-bg.png";
export const COMPANY_LOGO_URL =
  "http://www.zinqsolutions.com.au/wp-content/uploads/2018/12/zinq-comingsoon-logo2.png";
export const COMPANY_NAME = "Zuqo Solutions";
