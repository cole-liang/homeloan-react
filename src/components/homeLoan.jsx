import React, { Component } from "react";

import { Row, Col, Button } from "react-bootstrap";

import bannerBg from "../assets/bannerbg.jpg";
import styled from "styled-components";

const HomeLoanDiv = styled.div`
  height: 100%;
  background: url(${props => props.bannerBg}) no-repeat 82% bottom / cover;

  & .title {
    font-size: 40px;
  }

  & .content {
    margin-left: 50px;
  }
`;

class HomeLoan extends Component {
  state = {};

  render() {
    return (
      <HomeLoanDiv bannerBg={bannerBg}>
        <Col
          md={5}
          className="content noPadding h-100 d-flex flex-column justify-content-center"
        >
          <Row noGutters>
            <div className="title">We are helping people</div>
            <div className="title">
              <strong>just like you!</strong>
            </div>
          </Row>
          <Row noGutters>
            <div className="">I'm looking to</div>
          </Row>
          <Row noGutters>
            <Button>Get Started</Button>
          </Row>
        </Col>
      </HomeLoanDiv>
    );
  }
}

export default HomeLoan;
