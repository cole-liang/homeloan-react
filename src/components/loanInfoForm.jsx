import React from "react";
import BasicForm from "./common/basicForm";
import SectionWrapper from "./common/sectionWrapper";

import { Formik } from "formik";
import { Form, Row, Col } from "react-bootstrap";

import styled from "styled-components";

import * as yup from "yup";

const LoanInfoFormDiv = styled.div`
  margin-top: 20px;

  & .form {
    margin-bottom: -30px;
  }
`;

const incomeTypeError = "Income should be numbers";
const incomeGreaterThanZeroError = "Income should be greater than 0";
const incomeRequiredError = "Income is required";

const expenseTypeError = "Expense should be numbers";
const expenseGreaterThanZeroError = "Expense should be greater than 0";
const expenseRequiredError = "Expense is required";

const propertyValueTypeError = "Property value should be numbers";
const propertyValueGreaterThanZeroError =
  "Property value should be greater than 0";
const propertyValueRequiredError = "Property value is required";

const depositTypeError = "Deposit should be numbers";
const depositGreaterThanZeroError = "Deposit should be greater than 0";
const depositRequiredError = "Deposit is required";

const postcodeRegex = /^\d{4}$/;
const postcodeValidError = "Postcode should be four digits";
const postcodeRequiredError = "Postcode is required";

let schema = yup.object({
  yourIncome: yup
    .number()
    .typeError(incomeTypeError)
    .moreThan(0, incomeGreaterThanZeroError)
    .required(incomeRequiredError),
  yourPeriod: yup.string().required(),
  yourExpense: yup
    .number()
    .typeError(expenseTypeError)
    .moreThan(0, expenseGreaterThanZeroError)
    .required(expenseRequiredError),
  partnerIncome: yup.number().notRequired(),
  partnerPeriod: yup.string().notRequired(),
  partnerExpense: yup.number().notRequired(),
  propertyValue: yup
    .number()
    .typeError(propertyValueTypeError)
    .moreThan(0, propertyValueGreaterThanZeroError)
    .required(propertyValueRequiredError),
  deposit: yup
    .number()
    .typeError(depositTypeError)
    .moreThan(0, depositGreaterThanZeroError)
    .required(depositRequiredError),
  postcode: yup
    .string()
    .matches(postcodeRegex, postcodeValidError)
    .required(postcodeRequiredError)
});

export class LoanInfoForm extends BasicForm {
  state = {
    initialValues: {
      yourIncome: "",
      yourPeriod: "weekly",
      yourExpense: "",
      partnerIncome: "",
      partnerPeriod: "weekly",
      partnerExpense: "",
      propertyValue: "",
      deposit: "",
      postcode: ""
    }
  };

  componentWillMount() {
    const { isCouple } = this.props;
    if (isCouple) {
      schema = schema.shape({
        partnerIncome: yup
          .number()
          .typeError(incomeTypeError)
          .moreThan(0, incomeGreaterThanZeroError)
          .required(incomeRequiredError),
        partnerPeriod: yup.string().required(),
        partnerExpense: yup
          .number()
          .typeError(expenseTypeError)
          .moreThan(0, expenseGreaterThanZeroError)
          .required(expenseRequiredError)
      });
    }
  }

  handleSubmit = data => {
    this.props.onSubmit(data);
  };

  render() {
    const { initialValues } = this.state;
    const { isCouple, previousStep } = this.props;

    return (
      <LoanInfoFormDiv>
        <Col xs={12} md={11} className="m-auto noPadding">
          <Row className="justify-content-between">
            <Col xs={12}>
              <Formik
                validationSchema={schema}
                onSubmit={this.handleSubmit}
                initialValues={initialValues}
              >
                {({
                  handleSubmit,
                  handleChange,
                  handleBlur,
                  values,
                  touched,
                  isValid,
                  errors
                }) => {
                  const commonAttrs = [
                    values,
                    touched,
                    errors,
                    handleChange,
                    handleBlur
                  ];
                  return (
                    <Form noValidate onSubmit={handleSubmit}>
                      <Row className="justify-content-center">
                        <Col xs={12} xl={8} className="form">
                          <SectionWrapper name="About You">
                            <Row className="justify-content-center">
                              <Col xs={12} lg={7}>
                                {this.renderInput(
                                  "yourIncome",
                                  ...commonAttrs,
                                  {
                                    label: "Your Income",
                                    prepend: "AUD",
                                    required: true,
                                    periodSelect: this.renderSelect(
                                      "yourPeriod",
                                      handleChange,
                                      [
                                        "weekly",
                                        "fortnightly",
                                        "monthly",
                                        "quarterly",
                                        "yearly"
                                      ]
                                    )
                                  }
                                )}
                              </Col>
                              <Col xs={12} lg={5}>
                                {this.renderInput(
                                  "yourExpense",
                                  ...commonAttrs,
                                  {
                                    label: "Your Monthly Expense",
                                    prepend: "AUD",
                                    required: true
                                  }
                                )}
                              </Col>
                              {isCouple && (
                                <React.Fragment>
                                  <Col xs={12} lg={7}>
                                    {this.renderInput(
                                      "partnerIncome",
                                      ...commonAttrs,
                                      {
                                        label: "Partner's Income",
                                        prepend: "AUD",
                                        required: true,
                                        periodSelect: this.renderSelect(
                                          "partnerPeriod",
                                          handleChange,
                                          [
                                            "weekly",
                                            "fortnightly",
                                            "monthly",
                                            "quarterly",
                                            "yearly"
                                          ]
                                        )
                                      }
                                    )}
                                  </Col>
                                  <Col xs={12} lg={5}>
                                    {this.renderInput(
                                      "partnerExpense",
                                      ...commonAttrs,
                                      {
                                        label: "Partner's Monthly Expense",
                                        prepend: "AUD",
                                        required: true
                                      }
                                    )}
                                  </Col>
                                </React.Fragment>
                              )}
                              <Col xs={12}>
                                {this.renderInput("deposit", ...commonAttrs, {
                                  label: "Your Deposit",
                                  prepend: "AUD",
                                  required: true
                                })}
                              </Col>
                            </Row>
                          </SectionWrapper>
                          <SectionWrapper name="About Target Property">
                            <Row className="justify-content-center">
                              <Col xs={12} md={6}>
                                {this.renderInput(
                                  "propertyValue",
                                  ...commonAttrs,
                                  {
                                    label: "Property Value",
                                    prepend: "AUD",
                                    required: true
                                  }
                                )}
                              </Col>

                              <Col xs={12} md={6}>
                                {this.renderInput("postcode", ...commonAttrs, {
                                  label: "Postcode of the Property",
                                  required: true
                                })}
                              </Col>
                            </Row>
                          </SectionWrapper>
                          <Row
                            noGutters
                            className="justify-content-between  mt-4"
                          >
                            <Col xs={4} lg={3}>
                              {this.renderCustomButton("Back", () =>
                                previousStep()
                              )}
                            </Col>
                            <Col xs={4} lg={3}>
                              {this.renderSubmitButton("Next")}
                            </Col>
                          </Row>
                        </Col>
                      </Row>
                    </Form>
                  );
                }}
              </Formik>
            </Col>
          </Row>
        </Col>
      </LoanInfoFormDiv>
    );
  }
}

export default LoanInfoForm;
