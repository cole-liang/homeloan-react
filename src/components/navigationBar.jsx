import React from "react";
import { Nav, Navbar, NavDropdown } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";

const companyLogoPath =
  "http://www.zinqsolutions.com.au/wp-content/uploads/2018/12/zinq-comingsoon-logo2.png";

const NavigationBar = () => {
  return (
    <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
      <Navbar.Brand>
        <img src={companyLogoPath} width="30px" height="30px" />
        Zinq Solutions
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav className="mr-auto">
          <LinkContainer to="/">
            <Nav.Link>Home</Nav.Link>
          </LinkContainer>
          <LinkContainer to="/not-found">
            <Nav.Link>Get Started</Nav.Link>
          </LinkContainer>
          <NavDropdown title="Dropdown" id="collasible-nav-dropdown">
            <NavDropdown.Item href="#action/3.1">Action</NavDropdown.Item>
            <NavDropdown.Item href="#action/3.2">
              Another action
            </NavDropdown.Item>
            <NavDropdown.Item href="#action/3.3">Something</NavDropdown.Item>
            <NavDropdown.Divider />
            <NavDropdown.Item href="#action/3.4">
              Separated link
            </NavDropdown.Item>
          </NavDropdown>
        </Nav>
        <Nav>
          <LinkContainer to="/register">
            <Nav.Link>Register</Nav.Link>
          </LinkContainer>
          <LinkContainer to="/login">
            <Nav.Link>Login</Nav.Link>
          </LinkContainer>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};

export default NavigationBar;
