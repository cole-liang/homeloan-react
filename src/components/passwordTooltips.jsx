import React from "react";
import styled from "styled-components";

const PasswordToolTipsDiv = styled.div`
  & #pwdCharactersNumErrorText {
    color: ${props => (props.isPwdCharactersNumInvalid ? "red" : "green")};
  }

  & #pwdCharactersNumErrorIcon {
    margin-left: 30px;
  }

  & #pwdCharactersNumErrorIcon .fa-check-circle {
    display: ${props =>
      props.isPwdCharactersNumInvalid ? "none" : "inline-block"};
  }

  & #pwdCharactersNumErrorIcon .fa-times-circle {
    display: ${props =>
      props.isPwdCharactersNumInvalid ? "inline-block" : "none"};
  }

  & #pwdAlphaDigitErrorText {
    color: ${props => (props.isPwdAlphaDigitInvalid ? "red" : "green")};
  }

  & #pwdAlphaDigitErrorIcon {
    margin-left: 16px;
  }

  & #pwdAlphaDigitErrorIcon .fa-check-circle {
    display: ${props =>
      props.isPwdAlphaDigitInvalid ? "none" : "inline-block"};
  }

  & #pwdAlphaDigitErrorIcon .fa-times-circle {
    display: ${props =>
      props.isPwdAlphaDigitInvalid ? "inline-block" : "none"};
  }

  & #pwdTypeErrorText {
    color: ${props => (props.isPwdTypeInvalid ? "red" : "green")};
  }

  & #pwdTypeErrorIcon {
    margin-left: 53px;
  }

  & #pwdTypeErrorIcon .fa-check-circle {
    display: ${props => (props.isPwdTypeInvalid ? "none" : "inline-block")};
  }

  & #pwdTypeErrorIcon .fa-times-circle {
    display: ${props => (props.isPwdTypeInvalid ? "inline-block" : "none")};
  }

  & i {
    font-size: 17px;
  }

  & i.fa-check-circle {
    color: green;
  }

  & i.fa-times-circle {
    color: red;
  }
`;

const PasswordToolTips = ({ possibleErrors, passwordErrors }) => {
  return (
    <PasswordToolTipsDiv
      isPwdCharactersNumInvalid={passwordErrors.includes(
        possibleErrors.pwdCharactersNumError
      )}
      isPwdAlphaDigitInvalid={passwordErrors.includes(
        possibleErrors.pwdAlphaDigitError
      )}
      isPwdTypeInvalid={passwordErrors.includes(possibleErrors.pwdTypeError)}
    >
      {Object.keys(possibleErrors).map(error => (
        <p key={`${error}Text`} id={`${error}Text`}>
          {possibleErrors[error]}
          <span id={`${error}Icon`}>
            <i className="fa fa-check-circle" />
            <i className="fa fa-times-circle" />
          </span>
        </p>
      ))}
    </PasswordToolTipsDiv>
  );
};

export default PasswordToolTips;
