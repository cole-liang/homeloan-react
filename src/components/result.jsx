import React from "react";

import { Alert } from "antd";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { Row, Col } from "react-bootstrap";

const Result = ({ canHelp, user }) => {
  const successTip = !!user
    ? `Congratulation ${user.firstName}, we can help you!`
    : "Congratulation! We can help you!";
  const successMsg = !!user ? (
    <div>
      Please follow the <Link to="/">instructions</Link> to get your home loan.
    </div>
  ) : (
    <div>
      It is easy to get more home loan for you with just one further step:{" "}
      <Link to="/register">Click to register</Link>
    </div>
  );

  const warningTip = !!user
    ? `Hey ${user.firstName}, we need more information to help you...`
    : "We need more information to help you...";
  const warningMsg = (
    <div>
      <div>Please give us a call for further assistance.</div>
      <div>
        Or you may want to <Link to="/getStarted/1">start over</Link> again to
        change your info
      </div>
    </div>
  );

  return (
    <Col xs={12} className="d-flex align-items-center">
      <Row className="justify-content-center w-100">
        <Col xs={11}>
          {canHelp && (
            <Alert
              message={successTip}
              description={successMsg}
              type="success"
              showIcon
            />
          )}
          {!canHelp && (
            <Alert
              message={warningTip}
              description={
                <div>
                  <div>{warningMsg}</div>
                </div>
              }
              type="warning"
              showIcon
            />
          )}
        </Col>
      </Row>
    </Col>
  );
};

const mapStateToProps = state => {
  return {
    user: state.user
  };
};

export default connect(
  mapStateToProps,
  null
)(Result);
