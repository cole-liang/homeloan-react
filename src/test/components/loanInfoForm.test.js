import React from "react";
import { shallow } from "enzyme";
import { LoanInfoForm } from "./../../components/loanInfoForm";

describe("LoanInfoForm Component Test", () => {
  const previousStep = jest.fn();
  const onSubmit = jest.fn();
  const props = { isCouple: true, previousStep, onSubmit };

  it("should render LoanInfoForm component", () => {
    const wrapper = shallow(<LoanInfoForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  // it("should render LoanInfoForm component with errors with no data", () => {
  //   const wrapper = shallow(<LoanInfoForm {...props} />);
  //   const formikWrapper = wrapper.find("Formik");
  //   const innerFormWrapper = formikWrapper
  //     .dive()
  //     .find("Form")
  //     .dive()
  //     .find("form");
  //   // .dive()
  //   // .find("Form")
  //   // .dive()
  //   // .find("form");

  //   // expect(formWrapper).toMatchSnapshot();

  //   innerFormWrapper.simulate("submit", {
  //     preventDefault: () => {}
  //   });
  //   // formWrapper
  //   //   .find('Button[type="submit"]')
  //   //   .dive()
  //   //   .simulate("click");

  //   // expect(formWrapper.find('Button[type="submit"]').dive()).toMatchSnapshot();
  //   expect(onSubmit).toHaveBeenCalled();
  //   // wrapper
  //   //   .dive()
  //   //   .find("button")
  //   //   .at(1)
  //   //   .simulate("click");

  //   expect(
  //     innerFormWrapper.find('FormControl[name="yourIncome"]').dive()
  //   ).toMatchSnapshot();

  //   // expect(
  //   //   formikWrapper
  //   //     .find('FormControl[name="yourIncome"]')
  //   //     .dive()
  //   //     .hasClass("is-invalid")
  //   // ).to.equal(true);
  // });

  // it("should render LoanInfoForm component without errors", () => {
  //   const wrapper = shallow(<LoanInfoForm {...props} />);
  //   expect(wrapper).toMatchSnapshot();
  // });

  // it("onSubmit should be called when clicking Next button", () => {
  //   const wrapper = shallow(<LoanInfoForm {...props} />);
  //   expect(wrapper).toMatchSnapshot();
  // });

  // it("should render LoanInfoForm component", () => {
  //   const wrapper = shallow(<LoanInfoForm {...props} />);
  //   expect(wrapper).toMatchSnapshot();
  // });

  // it("should render LoanInfoForm component", () => {
  //   const wrapper = shallow(<LoanInfoForm {...props} />);
  //   expect(wrapper).toMatchSnapshot();
  // });
});
