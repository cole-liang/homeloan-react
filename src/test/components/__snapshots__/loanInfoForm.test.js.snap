// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanInfoForm Component Test should render LoanInfoForm component 1`] = `
<styled.div>
  <Col
    as="div"
    className="m-auto noPadding"
    md={11}
    xs={12}
  >
    <Bootstrap(Row)
      className="justify-content-between"
    >
      <Col
        as="div"
        xs={12}
      >
        <Formik
          enableReinitialize={false}
          initialValues={
            Object {
              "deposit": "",
              "partnerExpense": "",
              "partnerIncome": "",
              "partnerPeriod": "weekly",
              "postcode": "",
              "propertyValue": "",
              "yourExpense": "",
              "yourIncome": "",
              "yourPeriod": "weekly",
            }
          }
          isInitialValid={false}
          onSubmit={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
          validationSchema={
            ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": undefined,
              "_nodes": Array [
                "postcode",
                "deposit",
                "propertyValue",
                "partnerExpense",
                "partnerPeriod",
                "partnerIncome",
                "yourExpense",
                "yourPeriod",
                "yourIncome",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "deposit": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": true,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "partnerExpense": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": true,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "partnerIncome": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": true,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "partnerPeriod": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": true,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "postcode": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": true,
                    "undefined": false,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "propertyValue": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": true,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "yourExpense": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": true,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "yourIncome": NumberSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": true,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "number",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "yourPeriod": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": true,
                  },
                  "_mutate": undefined,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            }
          }
        >
          <Component />
        </Formik>
      </Col>
    </Bootstrap(Row)>
  </Col>
</styled.div>
`;
